frontend/src/pages/Listing/Listing.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useContext, useEffect, useState } from "react";
+
+import style from "./Listing.module.css";
+import { UserLoginContext } from "src/context/userLoginContext";
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Box,
+  Button,
+  CardMedia,
+  Typography,
+} from "@mui/material";
+import PageHeader from "src/components/PageHeader/PageHeader";
+import { useNavigate } from "react-router-dom";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import AddToCartButton from "src/components/Listing/AddToCartButton";
+import { useHttpClient } from "src/hooks/http-hook";
+import { API_BASE_URL, LISTING } from "src/constants/endpoints";
+import RecommendedCarousel from "src/components/Recommended/RecommendedCarousel";
+import { IListingCardProps } from "src/constants/common";
+
+const Listing = () => {
+  const { isLoggedIn } = useContext(UserLoginContext);
+  const navigate = useNavigate();
+
+  const [currentListing, setCurrentListing] = useState<IListingCardProps>();
+
+  const listingId = window.location.href.split("/").at(-1) || "";
+
+  const loginAndRedirectBackHere = () => {
+    localStorage.setItem("amazin_redirectListing", listingId);
+    navigate("/login");
+  };
+
+  const { sendRequest } = useHttpClient();
+  useEffect(() => {
+    const getSellerItems = async () => {
+      let item: IListingCardProps = await sendRequest(
+        LISTING,
+        "GET",
+        {},
+        `/${listingId}`
+      );
+      setCurrentListing(item);
```

Comment:
Calling setState inside useEffect is causing an infinite loop of updating. We need to add a dependency array

Reply:
Addressed, empty array added